ocaml/xapi/xapi_pif.ml

```diff
@@ -709,7 +709,14 @@ let reconfigure_ipv6 ~__context ~self ~mode ~iPv6 ~gateway ~dNS =
   Db.PIF.set_ipv6_configuration_mode ~__context ~self ~value:mode ;
   Db.PIF.set_ipv6_gateway ~__context ~self ~value:gateway ;
   Db.PIF.set_IPv6 ~__context ~self ~value:[iPv6] ;
-  if dNS <> "" then Db.PIF.set_DNS ~__context ~self ~value:dNS ;
+  (* Keep IPv4 DNS entries *)
+  let pif_dns =
```

Comment:
What if `dNS` is already in `pif_dns`? We probably don't want to add it to appear twice in the database

Reply:
This can't happen since when setting dns in IPv6 we only keep IPv4 dns entries and same thing the other way.

And the method check above in the code that the new dns entries are in the correct ip family.

But anyway this is still not good enough to work when using both v4 and v6 I've opened [an issue](https://github.com/xapi-project/xen-api/issues/4301) to find a better way of doing it, I proposed to add a new field for IPv6 dns.